components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1675,10 +1675,32 @@ bool RewardsServiceImpl::ShouldShowOnboarding() const {
 void RewardsServiceImpl::SaveOnboardingResult(OnboardingResult result) {
   PrefService* prefs = profile_->GetPrefs();
   prefs->SetTime(prefs::kOnboarded, base::Time::Now());
-  if (result == OnboardingResult::kOptedIn) {
-    SetAutoContributeEnabled(true);
-    SetAdsEnabled(true);
+
+  if (result != OnboardingResult::kOptedIn) {
+    return;
+  }
+
+  if (!IsInitialized()) {
```

Comment:
The `IsInitialized` check might be problematic, because it will return false if the process has been started but ledger initialization is not complete.

I'm wondering if we can't use `ready_` here to help solve the general problem here; will ping with some ideas.

Reply:
Good point.

Reply:
Rebased on `origin/ksmith-start-process` to incorporate changes @zenparsing made (thanks!) to address this issue.

Reply:
With the changes `StartProcess`, you should be able to skip the `IsInitialized` check now and just always call `StartProcess` (it will execute the callback if everything is already initialized).

You might also be able to get rid of the `CompleteOnboarding` method and just roll that logic into `OnStartProcessForOnboarding`.

Reply:
Yes, great point. Fixed.